src/calculation/CompositeReportBuilder.ts

```diff
@@ -70,6 +89,71 @@ export class CompositeReportBuilder<T extends PopulationGroupResult> extends Abs
     };
   }
 
+  public addResults(results: ExecutionResult<T>[]) {
+    // the weighted composite scoring type is component-based, but the others are individual-based
+    if (this.compositeScoringType === 'weighted') {
+      this.addComponentResults(results);
+    } else {
+      results.forEach(result => {
+        this.addPatientResults(result);
+      });
+    }
+  }
+
+  private addComponentResults(results: ExecutionResult<T>[]) {
```

Comment:
maybe we should rename this to addWeightedResults or something since it's specific to that